src/components/k8s/pcn_k8s/networkpolicies/default_policy_parser.go

```diff
@@ -0,0 +1,958 @@
+		//	Protocol & Port
+		//-------------------------------------
+
+		//	First, parse the protocol: so that if an unsupported protocol is listed, we silently ignore it.
+		//	By doing it this way we don't have to remove rules later on
+		if len(rule.Ports) > 0 {
+			generatedPorts = d.ParsePorts(rule.Ports)
+
+			//	If this rule consists of only unsupported protocols, then we can't go on!
+			//	If we did, we would be creating wrong rules!
+			//	We just need to ignore the rules, for now.
+			//	But if there is at least one supported protocol, then we can proceed
+			if len(generatedPorts) == 0 {
+				proceed = false
+			}
+		}
+
+		//-------------------------------------
+		//	Peers
+		//-------------------------------------
+
+		//	To is {} ?
+		if rule.To == nil && proceed {
+			result := d.GetConnectionTemplate(direction, "", "", pcn_types.ActionForward, []pcn_types.ProtoPort{})
+			generatedIngressRules = append(generatedIngressRules, result.Ingress...)
+			generatedEgressRules = append(generatedEgressRules, result.Egress...)
+		}
+
+		for i := 0; rule.To != nil && i < len(rule.To) && proceed; i++ {
+			to := rule.To[i]
+
+			//	IPBlock?
+			if to.IPBlock != nil {
+				ipblock := d.ParseIPBlock(to.IPBlock, direction)
+				generatedIngressRules = append(generatedIngressRules, ipblock.Ingress...)
+				generatedEgressRules = append(generatedEgressRules, ipblock.Egress...)
+			}
+
+			//	PodSelector Or NamespaceSelector?
+			if to.PodSelector != nil || to.NamespaceSelector != nil {
+				rulesGot, err := d.ParseSelectors(to.PodSelector, to.NamespaceSelector, namespace, direction)
+
+				if err == nil {
+					if len(rulesGot.Ingress) > 0 {
+						generatedIngressRules = append(generatedIngressRules, rulesGot.Ingress...)
+					}
+					if len(rulesGot.Egress) > 0 {
+						generatedEgressRules = append(generatedEgressRules, rulesGot.Egress...)
+					}
+				} else {
+					l.Errorln("Error while parsing selectors:", err)
+				}
+			}
+		}
+
+		//-------------------------------------
+		//	Finalize
+		//-------------------------------------
+		rulesWithPorts := d.insertPorts(generatedIngressRules, generatedEgressRules, generatedPorts)
+		parsed.Ingress = append(parsed.Ingress, rulesWithPorts.Ingress...)
+		parsed.Egress = append(parsed.Egress, rulesWithPorts.Egress...)
+	}
+
+	return parsed
+}
+
+// insertPorts will complete the rules by adding the appropriate ports
+func (d *DefaultPolicyParser) insertPorts(generatedIngressRules, generatedEgressRules []k8sfirewall.ChainRule, generatedPorts []pcn_types.ProtoPort) pcn_types.ParsedRules {
+
+	//	Don't make me go through this if there are no ports
+	if len(generatedPorts) < 1 {
+		return pcn_types.ParsedRules{
+			Ingress: generatedIngressRules,
+			Egress:  generatedEgressRules,
+		}
+	}
+
+	parsed := pcn_types.ParsedRules{
+		Ingress: []k8sfirewall.ChainRule{},
+		Egress:  []k8sfirewall.ChainRule{},
+	}
+
+	var waitForChains sync.WaitGroup
+	waitForChains.Add(2)
+
+	go func() {
```

Comment:
I totally agree on this and it could probably be removed. 
I just wondered if it was better to do it concurrently in case lots of pods have been found (e.g.: if 100 pods are found, 2 rules are generated for each, one that accepts new connections and one that accepts established. Multiplied by 2 because it is done for their virtual IP as well => 400 rules in total for one chain.). 
But since the computation is trivial it may be done sequentially.
Maybe I can also try to do this with fixed allocation instead of doing `append`